### PR TITLE
Refactor 10 holograph tests

### PR DESCRIPTION
**Pull request by mmvazzano**
## Describe Changes

 Refactor 10_holograph_tests.t.sol
- Run prettier
- Add a reference to the hardhat tests in the comments
- Replace some constants and error messages with the ones in Constants.sol file
- Add 'view' to some tests

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] All Foundry tests are passing

### CODE REVIEW

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 1]
```diff
@@ -2,698 +2,759 @@
...
+
+  /**
+   * @notice Test the setAdmin function of the Holograph contract.
+   * @dev This test is designed to verify the functionality of setting a new admin address in the contract.
+   * This test retrieves the current admin address from the Holograph contract,
+   * performs a prank operation on the admin address using the VM,
+   * and then sets a new admin address to the contract.
+   * Refers to the hardhat test with the description 'should successfully init once'
+   */
+  function testSetAdmin() public {
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol:102]: I would add a call to the current admin to validate that it is set up correctly.

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 2]
```diff
@@ -2,698 +2,759 @@
...
+  /* -------------------------------------------------------------------------- */
+
+  /**
+   * @notice Test the ability of the admin to alter the bridge slot in the Holograph contract.
+   * @dev This test is designed to verify that the admin can alter the bridge slot in the contract.
+   * This test performs a prank operation on the current admin address of the Holograph contract,
+   * and then sets a new random address as the bridge in the contract.
+   * Refers to the hardhat test with the description 'should allow admin to alter _bridgeSlot'
+   */
+  function testAllowAdminAlter_bridgeSlot() public {
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol]: remove the snake case

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 3]
```diff
@@ -2,698 +2,759 @@
...
+  }
+
+  /**
+   * @notice Test the revert behavior when the owner tries to alter the bridge slot in the Holograph contract.
+   * @dev This test is designed to verify that only the admin can alter the bridge slot in the contract.
+   * This test expects a revert with the message 'HOLOGRAPH: admin only function' when the owner attempts to
+   * alter the bridge slot in the Holograph contract.
+   * Refers to the hardhat test with the description 'should fail to allow owner to alter _bridgeSlot'
+   */
+  function testAllowOwnerToAlter_bridgeSlotRevert() public {
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol]: same here

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 4]
```diff
@@ -2,698 +2,759 @@
...
+   * @notice Test the ability of the admin to alter the bridge slot in the Holograph contract.
+   * @dev This test is designed to verify that the admin can alter the bridge slot in the contract.
+   * This test performs a prank operation on the current admin address of the Holograph contract,
+   * and then sets a new random address as the bridge in the contract.
+   * Refers to the hardhat test with the description 'should allow admin to alter _bridgeSlot'
+   */
+  function testAllowAdminAlter_bridgeSlot() public {
+    address random = RandomAddress.randomAddress();
+    vm.prank(origin);
+    holograph.setBridge(random);
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol:163]: same here, im callint to current bridge to verify the correct set

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 5]
```diff
@@ -2,698 +2,759 @@
...
+  }
+
+  /**
+   * @notice Test the revert behavior when a non-owner tries to alter the bridge slot in the Holograph contract.
+   * @dev This test is designed to verify that only the admin can alter the bridge slot in the contract.
+   * This test expects a revert with the message 'HOLOGRAPH: admin only function' when a non-owner attempts to
+   * alter the bridge slot in the Holograph contract.
+   * Refers to the hardhat test with the description 'should fail to allow non-owner to alter _bridgeSlot'
+   */
+  function testAllowNonOwnerToAlter_bridgeSlotRevert() public {
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol]: snake case

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 6]
```diff
@@ -2,698 +2,759 @@
...
+  /* -------------------------------------------------------------------------- */
+
+  /**
+   * @notice Test the validity of the chainId slot in the Holograph contract.
+   * @dev This test is designed to ensure the correct functionality of the chainId slot in the contract.
+   * This test verifies that the value returned by `getChainId()` in the Holograph contract
+   * is not equal to zero.
+   * Refers to the hardhat test with the description 'Should return valid _chainIdSlot'
+   */
+  function testReturnValid_chainIdSlot() public view {
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol]: snake case

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 7]
```diff
@@ -2,698 +2,759 @@
...
+  }
+
+  /* -------------------------------------------------------------------------- */
+  /*                                 GET CHAINID                                */
+  /* -------------------------------------------------------------------------- */
+
+  /**
+   * @notice Test the validity of the chainId slot in the Holograph contract.
+   * @dev This test is designed to ensure the correct functionality of the chainId slot in the contract.
+   * This test verifies that the value returned by `getChainId()` in the Holograph contract
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol]: return that the value set in initialization run ok

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 8]
```diff
@@ -2,698 +2,759 @@
...
+  /* -------------------------------------------------------------------------- */
+
+  /**
+   * @notice Test the ability of the admin to alter the chainId slot in the Holograph contract.
+   * @dev This test is designed to verify that the admin can alter the chainId slot in the contract.
+   * This test performs a prank operation on the current admin address of the Holograph contract,
+   * and then sets a new random address as the chainId in the contract.
+   * Refers to the hardhat test with the description 'should allow admin to alter _chainIdSlot'
+   */
+  function testAllowAdminAlter_chainIdSlot() public {
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol]: snake case

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 9]
```diff
@@ -2,698 +2,759 @@
...
+    vm.prank(origin);
+    holograph.setChainId((2));
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol:235]: same here, valid set its ok

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 10]
```diff
@@ -2,698 +2,759 @@
...
+   * @notice Test the ability of the admin to alter the factory slot in the Holograph contract.
+   * @dev This test is designed to verify that the admin can alter the factory slot in the contract.
+   * This test performs a prank operation on the current admin address of the Holograph contract,
+   * and then sets a new random address as the factory in the contract.
+   * Refers to the hardhat test with the description 'should allow admin to alter _factorySlot'
+   */
+  function testAllowAdminAlter_factorySlot() public {
+    address random = RandomAddress.randomAddress();
+    vm.prank(origin);
+    holograph.setFactory(random);
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol:307]: same here

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 11]
```diff
@@ -2,698 +2,759 @@
...
+  /**
+   * @notice Test the ability of the admin to alter the holographChainId slot in the Holograph contract.
+   * @dev This test is designed to verify that the admin can alter the holographChainId slot in the contract.
+   * This test performs a prank operation on the current admin address of the Holograph contract,
+   * and then sets a new random address as the holographChainId in the contract.
+   * Refers to the hardhat test with the description 'should allow admin to alter _holographChainIdSlot'
+   */
+  function testAllowAdminAlter_holographChainIdSlot() public {
+    vm.prank(origin);
+    holograph.setHolographChainId(2);
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol:380]: same here

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 12]
```diff
@@ -2,698 +2,759 @@
...
+   * @notice Test the ability of the admin to alter the interfaces slot in the Holograph contract.
+   * @dev This test is designed to verify that the admin can alter the interfaces slot in the contract.
+   * This test performs a prank operation on the current admin address of the Holograph contract,
+   * and then sets a new random address as the interfaces in the contract.
+   * Refers to the hardhat test with the description 'should allow admin to alter _interfacesSlot'
+   */
+  function testAllowAdminAlter_interfacesSlot() public {
+    address random = RandomAddress.randomAddress();
+    vm.prank(origin);
+    holograph.setInterfaces(random);
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol:452]: same here

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 13]
```diff
@@ -2,698 +2,759 @@
...
+   * @notice Test the ability of the admin to alter the operator slot in the Holograph contract.
+   * @dev This test is designed to verify that the admin can alter the operator slot in the contract.
+   * This test performs a prank operation on the current admin address of the Holograph contract,
+   * and then sets a new random address as the operator in the contract.
+   * Refers to the hardhat test with the description 'should allow admin to alter _operatorSlot'
+   */
+  function testAllowAdminAlter_operatorSlot() public {
+    address random = RandomAddress.randomAddress();
+    vm.prank(origin);
+    holograph.setOperator(random);
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol:526]: same here

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 14]
```diff
@@ -2,698 +2,759 @@
...
+   * @notice Test the ability of the admin to alter the registry slot in the Holograph contract.
+   * @dev This test is designed to verify that the admin can alter the registry slot in the contract.
+   * This test performs a prank operation on the current admin address of the Holograph contract,
+   * and then sets a new random address as the registry in the contract.
+   * Refers to the hardhat test with the description 'should allow admin to alter _registrySlot'
+   */
+  function testAllowAdminAlter_registrySlot() public {
+    address random = RandomAddress.randomAddress();
+    vm.prank(origin);
+    holograph.setRegistry(random);
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol:600]: same here

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 15]
```diff
@@ -2,698 +2,759 @@
...
+   * @notice Test the ability of the admin to alter the treasury slot in the Holograph contract.
+   * @dev This test is designed to verify that the admin can alter the treasury slot in the contract.
+   * This test performs a prank operation on the current admin address of the Holograph contract,
+   * and then sets a new random address as the treasury in the contract.
+   * Refers to the hardhat test with the description 'should allow admin to alter _treasurySlot'
+   */
+  function testAllowAdminAlter_treasurySlot() public {
+    address random = RandomAddress.randomAddress();
+    vm.prank(origin);
+    holograph.setTreasury(random);
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol:674]: same here

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 16]
```diff
@@ -2,698 +2,759 @@
...
+   * @notice Test the ability of the admin to alter the utilityToken slot in the Holograph contract.
+   * @dev This test is designed to verify that the admin can alter the utilityToken slot in the contract.
+   * This test performs a prank operation on the current admin address of the Holograph contract,
+   * and then sets a new random address as the utilityToken in the contract.
+   * Refers to the hardhat test with the description 'should allow admin to alter _utilityTokenSlot'
+   */
+  function testAllowAdminAlter_utilityTokenSlot() public {
+    address random = RandomAddress.randomAddress();
+    vm.prank(origin);
+    holograph.setUtilityToken(random);
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol:748]: same here

**Review by fgregoribattista — COMMENTED** [review]
Good work, we could improve it with the reviewed cases

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 1 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol:102]: Done!

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 2 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol]: Done

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 3 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol]: Done

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 4 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol:163]: Done

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 12 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol:452]: Done

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 13 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol:526]: Done

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 14 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol:600]: Done

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 15 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol:674]: Done

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 16 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol:748]: Done

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 7 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol:201]: Done

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 8 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol]: Done

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 9 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol:235]: Done

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 10 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol:307]: Done

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 11 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol:380]: Done

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 6 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol]: Done

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 7 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol]: Done

**Review by sogoiii — APPROVED** [review]
👍

**Review by lucas-marc — APPROVED** [review]
Looks good to me, ready to be merged

**Pull request merged**